lib.rs

```diff
@@ -68,9 +68,33 @@ fn log_to_slog_level(level: log::Level) -> Level {
     }
 }
 
+/// Gets a static string for Record location information.
+/// Uses a cache to avoid leaking each string more than once.
+#[cfg(feature = "record_location_unstable")]
+#[cached::proc_macro::cached]
+fn get_static_info(non_static_info: Option<String>) -> &'static str {
```

Comment:
Shouldn't this be `non_static_info: Option<&'str>`?

Reply:
The point of this function is to turn `&'a str` into `&'static str`

Reply:
Unfortunately it can't be. If I understand correctly, it's because it can't cache based on a non_static string as the cache key would be dropped. This is similar to issues I had with the interning libraries. 
So I have to clone it before it gets passed into this function. But it only does that clone in the case where it needs to (a static string doesn't already exist).